### PR TITLE
New observability + general refactor of http/grpc clients

### PR DESCRIPTION
**Pull request by labkode**
This PR:
- adds possibility to register prometheus collectors from any package and expose them thorough a metrics endpoint
- refactors packages dealing with context into just one `appctx`
- refactors usage of http.DefaultClient to custom client that will propagate tokens and trace ID

### CODE REVIEW

File: pkg/cbox/share/sql/sql.go [hunk 1]
```diff
@@ -0,0 +1,631 @@
+// Copyright 2018-2023 CERN
```
Comment (gmgigi96) [inline pkg/cbox/share/sql/sql.go:1]: remove this file

File: pkg/cbox/storage/eoshomewrapper/eoshomewrapper.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+// Copyright 2018-2023 CERN
```
Comment (gmgigi96) [inline pkg/cbox/storage/eoshomewrapper/eoshomewrapper.go:1]: remove this file

File: pkg/cbox/storage/eoswrapper/eoswrapper.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
+// Copyright 2018-2023 CERN
```
Comment (gmgigi96) [inline pkg/cbox/storage/eoswrapper/eoswrapper.go:1]: remove this file

**Review by gmgigi96 — CHANGES_REQUESTED** [review]

**Review by labkode — COMMENTED** [review]
changes done + added tests

**Review by gmgigi96 — APPROVED** [review]

**Pull request merged**